integration-cli/docker_cli_rmi_test.go

```diff
@@ -268,3 +268,43 @@ func (s *DockerSuite) TestRmiBlank(c *check.C) {
 		c.Fatalf("Expected error message not generated: %s", out)
 	}
 }
+
+func (s *DockerSuite) TestRmiContainerImageNotFound(c *check.C) {
+	// Build 2 image for testing.
+	images := []string{"test1", "test2"}
+	for _, name := range images {
+		cmd := exec.Command(dockerBinary, "build", "--no-cache", "-t", name, "-")
+		dockerFile := fmt.Sprintf("FROM busybox\nMAINTAINER %s\nRUN echo %s\n", name, name)
+		cmd.Stdin = strings.NewReader(dockerFile)
+
+		if out, _, err := runCommandWithOutput(cmd); err != nil {
+			c.Fatalf("Could not build %s: %s, %v", name, out, err)
+		}
+	}
+
+	// Create a long-running container.
+	cmd := exec.Command(dockerBinary, "run", "-d", images[0], "sleep", "300")
```

Comment:
And for the sake of over-communicating, should use `dockerCmd` in all the cases below as well.